server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -707,8 +708,23 @@ protected Symbol visitSubscriptExpression(SubscriptExpression node, ExpressionAn
                                                                  List.of(),
                                                                  operation,
                                                                  context.errorOnUnknownObjectKey());
-                        if (base instanceof Reference) {
-                            throw e;
+                        var baseType = ArrayType.unnest(base.valueType());
+                        if (baseType instanceof ObjectType objectType) {
+                            if (objectType.resolveInnerType(parts).id() != UndefinedType.ID) {
+                                return allocateFunction(
+                                    SubscriptFunction.NAME,
+                                    List.of(
+                                        node.base().accept(this, context),
+                                        node.index().accept(this, context)
+                                    ),
+                                    context
+                                );
+                            }
+                        }
```

Comment:
Aligned with this idea, we could validate whether it is a known key before allocating subscript functions.

At current state, the following case is not handled:

```
cr> create table qqq (o array(object as (o2 array(object))));                                                                                     
CREATE OK, 1 row affected  (1.783 sec)

cr> select o[1]['o2']::array(object)[1]['u'] from qqq;                                                                                            
+--------------------+
| o[1]['o2'][1]['u'] | -- 'u' unknown but empty results returned instead of throwing
+--------------------+
+--------------------+
SELECT 0 rows in set (4.212 sec)
```